pkg/cmd/template/schema_test.go

```diff
@@ -13,8 +13,115 @@ import (
 	"github.com/k14s/ytt/pkg/files"
 )
 
-func TestNullableScalarsAllowsNull(t *testing.T) {
-	schemaYAML := `#@schema/match data_values=True
+var opts *cmdtpl.TemplateOptions
+
+func TestMain(m *testing.M) {
+	opts = cmdtpl.NewOptions()
+	opts.SchemaEnabled = true
+
+	m.Run()
+}
+
+func TestDataValueConformingToSchemaSucceeds(t *testing.T) {
+	t.Run("Map Only", func(t *testing.T) {
+		schemaYAML := `#@schema/match data_values=True
+---
+db_conn:
+  hostname: ""
+  port: 0
+  username: ""
+  password: ""
+  metadata:
+    run: jobName
+  tls_only: false
+top_level: ""
+`
+		dataValuesYAML := `#@data/values
+---
+db_conn:
+  hostname: server.example.com
+  port: 5432
+  username: sa
+  password: changeme
+  metadata:
+    run: ./build.sh
+  tls_only: true
+top_level: key
+`
+		templateYAML := `---
+rendered: true`
+
+		testSchemaTemplates(t, schemaYAML, dataValuesYAML, templateYAML, "rendered: true\n")
+	})
+	t.Run("Map And Array", func(t *testing.T) {
+		schemaYAML := `#@schema/match data_values=True
+---
+db_conn:
+- hostname: ""
+  port: 0
+  username: ""
+  password: ""
+  metadata:
+    run: jobName
+top_level: ""
+`
+		dataValuesYAML := `#@data/values
+---
+db_conn:
+#@overlay/append
```

Comment:
Yes. Good news is that this is not a degradation in UX: before schemas, Configuration Authors have supplied a base Data Values that illustrated that a particular Data Value is a list. Configuration Consumers have already had to supply some kind of `@overlay` annotation on their arrays.

Even better news is that in the very next story, we're aiming to improve that UX further by setting a default behavior (likely `@overlay/replace` on the Array itself) so that, Configuration Consumers can provide configuration _without_ having to learn about overlays, first. 👍🏼   Stay tuned!